openedx/features/course_experience/templates/course_experience/course-home-fragment.html

```diff
@@ -106,6 +106,24 @@ <h3 class="vc-title vc-fade vc-polite-only">Pursue a verified certificate</h3>
                 % endif
             </main>
             <aside class="course-sidebar layout-col layout-col-a">
+                % if has_goal_permission:
+                    <div class="section section-goals ${'' if current_goal else 'hidden'}">
+                        <div class="current-goal-container">
+                            <span class="sr">${_("Set your course goal:")}</span>
+                            <h3 class="title hd-6">${_("Goal: ")}</h3>
+                            <div class="goal">
+                                <span class="text">${goal_options[current_goal.goal_key] if current_goal else ""}</span>
+                            </div>
+                            <select class="edit-goal-select">
+                                % for goal, goal_text in goal_options.items():
+                                    <option value="${goal}" ${"selected" if current_goal and current_goal.goal_key == goal else ""}>${goal_text}</option>
+                                % endfor
+                            </select>
+                            <span class="response-icon"></span>
+                            <button class="edit-icon fa fa-pencil" aria-hidden="true"></button>
```

Comment:
There should be screen reader text in here telling users they can edit their course goal.

Reply:
Added